pkg/rpcserver/rpcserver.go

```diff
@@ -82,7 +82,23 @@ type Server struct {
 	*runnerStats
 }
 
+// validateCfg checks required fields to be set
+// to avoid unexpected panics during Server creation.
+func validateCfg(cfg *mgrconfig.Config) error {
```

Comment:
This looks like a wrong place to validate manager config. We have pkg/mgrconfig with lots of validation already. How did you get a broken manager config here?